### ISSUE TITLE
[a11y]: ProgressIndicator causes horizontal scrollbar at 400% zoom

### ISSUE THREAD

**Issue by simon-burns** [carbon-design-system/carbon#18066]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Package version

v1.70.0

### Automated testing tool and ruleset

WCAG Mapping: 1.4.10 Reflow

### Description

When screen resized to 400% horizontal scroll bars are appearing on the page because the ProgressIndicator doesn't adapt to less horizontal space.

User impact: If a horizontal scroll bar appears when the page is reflowed to 320 pixels wide, it can make it difficult for users with visual impairments or other disabilities that require larger text sizes. This can make it difficult for these users to read and understand the content, leading to exclusion, frustration, and a negative user experience.

WCAG Mapping: 1.4.10 Reflow
Level of Compliance: AA

### WCAG 2.1 Violation

1.4.10 Reflow

### Reproduction/example

https://stackblitz.com/edit/github-s6er9y?file=src%2FApp.jsx

### Steps to reproduce

Reduce the horizontal space available for the Progress Indicator, either by reducing the width or by zooming in on the page.  You will see a horizontal scrollbar.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by szamoca-at-the-corp** [comment]
Hi,
We had to deal with this inside our application too. Our temporary, localised solution was to use [Cloud PAL's window resize hook](https://pages.github.ibm.com/ibmcloud/cloud-pal-community/?path=/docs/utilities-usewindowresize--docs) to keep track of the current screen width, and pass the `vertical` prop to the progress indicator component conditionally below a certain width.

Do you think this could be a generally good solution for the Carbon component itself, or is something else needed here?
